src/py/mat3ra/made/tools/analyze/interface.py

```diff
@@ -0,0 +1,72 @@
+from functools import cached_property
+
+import numpy as np
+from mat3ra.code.entity import InMemoryEntityPydantic
+from mat3ra.esse.models.core.abstract.matrix_3x3 import Matrix3x3Schema
+from mat3ra.esse.models.materials_category_components.entities.auxiliary.two_dimensional.supercell_matrix_2d import (
+    SupercellMatrix2DSchema,
+)
+from mat3ra.made.tools.build.slab.builders import SlabBuilder
+from mat3ra.made.tools.build.slab.configuration import SlabConfiguration, SlabStrainedSupercellConfiguration
+
+
+class InterfaceAnalyzer(InMemoryEntityPydantic):
+    substrate_slab_configuration: SlabConfiguration
+    film_slab_configuration: SlabConfiguration
+
+    @cached_property
+    def substrate_material(self):
```

Comment:
we should define 

`get_component_material(self, component_configuration)`
`get_substrate_material => get_component_material(self.substrate_configuration)`

Same for `substrate_strained_configuration` -> `get_component_strained_configuration`